python-package/setup.py

```diff
@@ -340,6 +340,7 @@ def run(self) -> None:
           version=version,
           description='LightGBM Python Package',
           long_description=readme,
+          python_requires='>=3.5',
```

Comment:
Thanks for this contribution!
But I don't think that LightGBM is compatible with Python 3.5 due to using f-strings e.g. here
https://github.com/microsoft/LightGBM/blob/15266f1d613a37fc6d59bc0462854a99eeee3b0b/python-package/setup.py#L134
that were introduced in Python 3.6, 
https://www.python.org/dev/peps/pep-0498/
```suggestion
          python_requires='>=3.6',
```